tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/PriceRepositoryTests.cs

```diff
@@ -63,10 +63,57 @@ public async Task GetPricesBySolutionIdQueryAsync_ValidSolutionId_ReturnsPriceLi
             request.Count().Should().Be(2);
         }
 
-        private static async Task InsertPriceAsync(string solutionId)
+        [Test]
+        public async Task GetPriceByPriceIdQueryAsync_InvalidPriceId_ReturnsEmptyList()
+        {
+            var request = await _priceRepository.GetPriceByPriceIdQueryAsync(-999, CancellationToken.None);
+
+            request.Should().BeEmpty();
+        }
+
+        [Test]
+        public async Task GetPricingByPriceIdQueryAsync_ValidPriceIdAndFlatPrice_ReturnsPriceListResult()
         {
-            await CataloguePriceEntityBuilder.Create()
+            int validPriceId = await InsertPriceAsync(_solutionId);
+
+            var request = await _priceRepository.GetPriceByPriceIdQueryAsync(validPriceId, CancellationToken.None);
+
+            request.Count().Should().Be(1);
+        }
+
+        [Test]
+        public async Task GetPricingByPriceIdQueryAsync_ValidPriceIdAndTiredPrice_ReturnsTieredPriceResult()
+        {
+            int validPriceId = await InsertTieredPriceAsync(_solutionId);
+            await InsertCalaloguePriceTier(validPriceId, 1, 2, 1.0m);
+            await InsertCalaloguePriceTier(validPriceId, 3, 4, 1.50m);
+            var response = await _priceRepository.GetPriceByPriceIdQueryAsync(validPriceId, CancellationToken.None);
+            response.Count(t => t.CataloguePriceTypeId == 2).Should().Be(2);
+        }
+
+        private static async Task<int> InsertPriceAsync(string solutionId)
+        {
+            return await CataloguePriceEntityBuilder.Create()
                 .WithCatalogueItemId(solutionId)
+                .Build().InsertAsync<int>();
+        }
+
+        private static async Task<int> InsertTieredPriceAsync(string solutionId)
+        {
+            var priceId = await CataloguePriceEntityBuilder.Create()
+                .WithPriceTypeId(2)
+                .WithCatalogueItemId(solutionId)
+                .Build().InsertAsync<int>();
+            return priceId;
+        }
+
+        private static async Task InsertCalaloguePriceTier(int priceId, int? startBand, int endBand, decimal price)
```

Comment:
your int? is on the wrong part. The endBand can be a nullable int but the start band shouldnt be. In one of your tests you should have a null endBand (the endBand is typically null for the last tier selected)